Sources/Algorithms/isSorted.swift

```diff
@@ -0,0 +1,51 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// isSorted()
+//===----------------------------------------------------------------------===//
+
+
+extension Sequence where Element: Comparable {
+    public func isSorted() -> Bool {
+        /// Returns Bool, indicating whether a sequence is sorted
+        /// into non-descending order.
+        ///
+        /// - Complexity: O(*n*), where *n* is the length of the sequence.
+        
+        isSorted(by: <)
+    }
+    
+    public func isSorted(by areInIncreasingOrder: (Element, Element) -> Bool) -> Bool {
```

Comment:
Two notes on this version: 

1. Since this method takes a closure for comparisons, it doesn’t need to be in the constrained extension
2. For this kind of method, we mark the closure as throwing and the method itself as `rethrows`